src/verify/VerificationStartScreen.tsx

```diff
@@ -121,23 +121,23 @@ function VerificationStartScreen({
     navigation.setOptions({
       headerTitle: title,
       headerRight: () =>
-        route.params?.isOnboarding && (
+        !route.params?.hasOnboarded && (
           <TopBarTextButton
             title={t('skip')}
             testID="PhoneVerificationSkipHeader"
             onPress={onPressSkip}
             titleStyle={{ color: colors.onboardingBrownLight }}
           />
         ),
-      headerLeft: () => !route.params?.isOnboarding && <BackButton />,
+      headerLeft: () => route.params?.hasOnboarded && <BackButton />,
       // Disable iOS back during onboarding
-      gestureEnabled: !route.params?.isOnboarding,
+      gestureEnabled: !!route.params?.hasOnboarded,
```

Comment:
nit: Does this need to be forced to a boolean?

```suggestion
      gestureEnabled: route.params?.hasOnboarded,
```

Reply:
it has to be forced to a boolean. The default value is `true`